### ISSUE TITLE
unable to control deconz group switch

### ISSUE THREAD

**Issue by ErnstEeldert** [ebaauw/homebridge-hue#315]
#### Issue
In an attempt to simplify my homekit setup, I am exposing groups of lights from deconz to homekit, and blacklisting the individual lights. The groups show up in home kit as expected, but I can't turn on/off the group. After a restart of homebridge (with the lights turned on), the light icon in the home app shows as on. If I tap it, it goes to off, but nothing happens to the lights. If I tap it again, the icon shows as on, the lights slowly flash a few times, and the icon goes back to off again. I have a number of groups defined like this, same behavior.

If I set the brightness of the group by long pressing the icon, the group responds accordingly. But once the new brightness has been set, the switch goes to off again.

#### Debug Files
[homebridge.log.gz](https://github.com/ebaauw/homebridge-hue/files/1821417/homebridge.log.gz)
[homebridge-hue.json.gz](https://github.com/ebaauw/homebridge-hue/files/1821418/homebridge-hue.json.gz)

**Comment by ebaauw** [comment]
I cannot find any debug logging of the interaction with the groups.  Did you turn on debugging with `homebridge -D` after interacting with the exposed groups?

Without the debug log of the interaction, I can only guess what's happening.  Could you un-blacklist the lights of one group (so I can also see what happens with the lights) and then flip that group's switch a couple of times (while running `homebridge -D`) and post the log?

> If I set the brightness of the group by long pressing the icon, the group responds accordingly. But once the new brightness has been set, the switch goes to off again.

That would suggest a hiccup in the communication between the deCONZ gateway and one of the lights in the group.  You have a lot of IKEA lights, so I wouldn't be surprised if this were the case.

Some other observations from the log:
```
[2018-3-17 08:54:46] [Hue] deconz: /sensors/23: warning: unknown ZHAPresence sensor {"config":{"battery":100,"duration":60,"on":true,"reachable":false,"temperature":2800},"ep":1,"etag":"42a51aebfc4f7504c74b92bce346fb4d","manufacturername":"LUMI","modelid":"lumi.sensor_wleak.aq1","name":"lumi.sensor_wleak.aq1","state":{"lastupdated":"2018-03-16T20:23:44","lowbattery":false,"presence":false,"tampered":false},"type":"ZHAPresence","uniqueid":"00:15:8d:00:02:13:05:63-01-0500"}
```
You might want to delete and re-pair the leak sensor, so it appears as a ZHAWater sensor.  homebridge-hue will then expose it properly.

```
[2018-3-17 08:54:47] [Hue] deconz: /sensors/28: Daylight "Daylight"
[2018-3-17 08:54:47] [Hue] deconz: /sensors/28: warning: Daylight sensor not configured
...
HAP Warning: Characteristic 00000027-0000-1000-8000-656261617577 not in required or optional characteristics for service 00000084-0000-1000-8000-0026BB765291. Adding anyway.
```
deCONZ should configure the Daylight sensor automatically, but this can take some time (hours?).  The HAP Warning is caused by homebridge-hue not handling the unconfigured Daylight sensor correctly.  It can be ignored.  I'll get rid of this warning in the next version of homebridge-hue.

**Comment by ErnstEeldert** [comment]
Thanks for the elaborate answer! I didn't get a notification for some reason, hence my slowish response.

I removed the 'Eethoek' lights from the blacklist, and had a look at the logs (homebridge is running in -D mode). Here's the only thing I can see when I flip the switch on the group:

Mar 18 09:16:06 raspy3 homebridge[12091]: [2018-3-18 09:16:06] [Hue] deconz: gateway request 27: put /groups/37/action {"alert":"select"}
Mar 18 09:16:06 raspy3 homebridge[12091]: [2018-3-18 09:16:06] [Hue] deconz: gateway request 27: ok

All group lights turn on shortly, seem to flash slightly, then turn off again. Just did a quick look at the deconz rest api docs, and that seems to be the correct behavior. Question is, why does it only send the select alert action to the group?

If I turn on an individual light, it looks like this, and the light turns on as expected:

Mar 18 09:21:14 raspy3 homebridge[12091]: [2018-3-18 09:21:14] [Hue] Eethoek - Spot 1: homekit on changed from 0 to 1
Mar 18 09:21:14 raspy3 homebridge[12091]: [2018-3-18 09:21:14] [Hue] deconz: gateway request 70: put /lights/48/state {"on":true}
Mar 18 09:21:14 raspy3 homebridge[12091]: [2018-3-18 09:21:14] [Hue] deconz: gateway request 70: ok
Mar 18 09:21:14 raspy3 homebridge[12091]: [2018-3-18 09:21:14] [Hue] Eethoek - Spot 1: state changed event
Mar 18 09:21:14 raspy3 homebridge[12091]: [2018-3-18 09:21:14] [Hue] Eethoek: state changed event
Mar 18 09:21:14 raspy3 homebridge[12091]: [2018-3-18 09:21:14] [Hue] Eethoek: group any_on changed from false to true
Mar 18 09:21:14 raspy3 homebridge[12091]: [2018-3-18 09:21:14] [Hue] Eethoek: set homekit any on from 0 to 1

**Comment by ebaauw** [comment]
That’s a bug in homebridge-hue allright, probably introduced while adding support for the Heiman Siren.  homebridge-hue should put `{“on”: true}`.

**Comment by ErnstEeldert** [comment]
That seems to be the case. Commented out line 968-970 from HueLight.js and the groups behave as expected.

**Comment by ebaauw** [comment]
The issue is that `this.config.on` wasn't set for groups.  Neither were `this.config.bri`, etc.  Because of this, the HomeKit characteristics for the group weren't updated when the group was changed from another Hue app.  Also found that `/groups/0` would always be exposed.

**Comment by DSchumacher2104** [comment]
I found the same problem here using the hue bridge, nit deCONZ. All group-Commands will produce "funny" light effects and don't work.

**Comment by ebaauw** [comment]
Yeah, it's a bug in homebridge-hue, irrespective of whether it connects to deCONZ or to a Hue bridge.  Will be fixed in next version due later today.

**Comment by ebaauw** [comment]
Released v0.8.7.
